tests/base/src/test/java/org/keycloak/tests/admin/model/workflow/AddRequiredActionTest.java

```diff
@@ -4,45 +4,60 @@
 import org.keycloak.models.UserModel;
 import org.keycloak.models.workflow.AddRequiredActionStepProvider;
 import org.keycloak.models.workflow.AddRequiredActionStepProviderFactory;
-import org.keycloak.models.workflow.UserCreationTimeWorkflowProviderFactory;
 import org.keycloak.representations.idm.UserRepresentation;
 import org.keycloak.representations.workflows.WorkflowStepRepresentation;
 import org.keycloak.representations.workflows.WorkflowRepresentation;
-import org.keycloak.testframework.annotations.InjectRealm;
 import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
-import org.keycloak.testframework.injection.LifeCycle;
-import org.keycloak.testframework.realm.ManagedRealm;
 import org.keycloak.testframework.realm.UserConfigBuilder;
 
 import java.util.List;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+import static org.keycloak.models.workflow.ResourceOperationType.USER_ADDED;
 
-@KeycloakIntegrationTest(config = WorkflowsServerConfig.class)
-public class AddRequiredActionTest {
-
-    @InjectRealm(lifecycle = LifeCycle.METHOD)
-    ManagedRealm managedRealm;
+@KeycloakIntegrationTest(config = WorkflowsBlockingServerConfig.class)
+public class AddRequiredActionTest extends AbstractWorkflowTest {
 
     @Test
     public void testStepRun() {
         managedRealm.admin().workflows().create(WorkflowRepresentation.create()
-                .of(UserCreationTimeWorkflowProviderFactory.ID)
-                .name(UserCreationTimeWorkflowProviderFactory.ID)
+                .name("myworkflow")
+                .onEvent(USER_ADDED.name())
                 .withSteps(
                         WorkflowStepRepresentation.create()
                                 .of(AddRequiredActionStepProviderFactory.ID)
                                 .withConfig(AddRequiredActionStepProvider.REQUIRED_ACTION_KEY, "UPDATE_PASSWORD")
                                 .build()
                 ).build()).close();
 
-        managedRealm.admin().users().create(UserConfigBuilder.create().username("test").build()).close();
+        managedRealm.admin().users().create(UserConfigBuilder.create().username("myuser").build()).close();
+
+        List< UserRepresentation> users;
+        UserRepresentation userRepresentation;
+
+        int i = 0;
+        while (true) {
```

Comment:
you could use the Awaitility library (change the timeout and poll interval to the desired values)
```java
        Awaitility.await()
                .timeout(Duration.ofSeconds(30))
                .pollInterval(Duration.ofSeconds(1))
                .untilAsserted(() -> {
                    var users = managedRealm.admin().users().search("myuser");
                    assertThat(users, hasSize(1));
                    var userRepresentation = users.get(0);
                    Assertions.assertTrue(userRepresentation.getRequiredActions() != null && !userRepresentation.getRequiredActions().isEmpty());
                });
```